### PR TITLE
Can now save reservations

### PR DESCRIPTION
**Pull request by Anelon**
Reservations are now actually saved, makes new customer if customer didn't exist already (both rest and standard controller)

### CODE REVIEW

File: src/main/java/cst438flights/controller/flightRestController.java [hunk 1]
```diff
@@ -2,18 +2,20 @@
...
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
 @RestController
 public class flightRestController {
+    //for holding the origin name
+    static final String ORIGIN = "planner";
```
Comment (espresso64) [inline src/main/java/cst438flights/controller/flightRestController.java:18]: Adjusted ORIGIN to equal "planner", now matches reservationOrigin term used in other services.

File: src/main/java/cst438flights/controller/flightRestController.java [hunk 2]
```diff
@@ -45,4 +47,39 @@
...
+        Reservation reservation = flightService.requestReservation(email, seatClass, numPassengers, prioBoarding, ORIGIN, flightID);
+
+        if (reservation == null) {
+            return new ResponseEntity<Reservation>(HttpStatus.NOT_FOUND);
+        } else {
+            return new ResponseEntity<Reservation>(reservation, HttpStatus.OK);
+        }
+    }
+    
+    //getMapping option to book reservation
```
Comment (espresso64) [inline src/main/java/cst438flights/controller/flightRestController.java:69]: Added a getMapping option. Gives planner team the use option and tests logic of PostMapping

**Review by espresso64 — APPROVED** [review]
*Deleted, re-approved down below.

File: src/main/java/cst438flights/controller/flightRestController.java (hunk 2 above)
Comment (espresso64) [reply src/main/java/cst438flights/controller/flightRestController.java:69]: Tested to work in Postman

File: src/main/resources/templates/index.html [hunk 3]
```diff
@@ -34,19 +34,21 @@ <h3>Select your departing city</h3> <br />
 
     <select name="seatClass" id="classSelect" required>
         <option value="">Select Seat Class</option>
-        <option value="firstClass" id="firstClassSeats">First Class $300, Seats Available: x</option>
-        <option value="businessClass" id="businessClassSeats">Business Class $200, Seats Available: x</option>
-        <option value="economyClass" id="economyClassSeats">Economy Class $100, Seats Available: x</option>
+        <option value="first" id="firstClassSeats">First Class $300, Seats Available: x</option>
```
Comment (espresso64) [inline src/main/resources/templates/index.html:37]: Adjusted the option values to match the ones used in the flight service. Added a small description to the priority boarding pricing.

**Review by espresso64 — APPROVED** [review]
Added some minor changes and enhancements. Reservations + pricing saving correctly to database, both through the website and rest. Overall, very nice job on implementing the reservation section! Approved.

File: src/main/java/cst438flights/service/FlightService.java [hunk 4]
```diff
@@ -46,13 +50,40 @@ public FlightInfo getFlightInfo(String arrivalairport) {
...
-                                   int numPassengers, boolean priorityBoarding,
-                                   String origin, int flightID
+    public Reservation requestReservation(String email, String seatClass,
+                                          int numPassengers, boolean priorityBoarding,
+                                          String origin, int flightID
     ) {
-        String msg = "{\"departureDate\": \"" + departureAirport + "\" \"arrivalAirport\": \"" + arrivalAirport + "\" \"departureDate\": \"" + departureDate + "\"}";
-        System.out.println("Sending message:" + msg);
-        rabbitTemplate.convertSendAndReceive(fanout.getName(), "", msg);
+        //some sanity checks
```
Comment (Nhinds88) [inline src/main/java/cst438flights/service/FlightService.java:57]: Good work on the request reservation.

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # DB
 # Hibernate
 spring.datasource.url = jdbc:mysql://localhost:3306/flights
-spring.datasource.username = root
-spring.datasource.password= otterotter
+spring.datasource.username = person
```
Comment (Nhinds88) [inline src/main/resources/application.properties:4]: We'll add the jawsdb credentials here after this merge

**Review by Nhinds88 — APPROVED** [review]

**Pull request merged**